tools/misc/mapping_extractor.py

```
@@ -0,0 +1,259 @@
+        
+        # Check number of columns with same name
+        count = raw_header_counts.get(column_name, 0)
+        
+        if count == 0:
+            msg = f"Target column \"{column_name}\" not found in the Excel sheet."
+            print(f"⚠️  [WARNING] {msg}")
+            warnings_list.append(msg)
+        elif count > 1:
+            msg = f"Column name \"{column_name}\" appears {count} times in the sheet. The exported column may not be the one expected."
+            print(f"⚠️  [WARNING] {msg}")
+            print(f"💡 Tip: Consider renaming the column in Excel or specifying a unique name in the YAML.")
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)